campaign-launcher/interfaceV2/src/components/modals/EditApiKeyModal/index.tsx

```diff
@@ -0,0 +1,208 @@
+import { FC, useEffect } from 'react';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  Box,
+  Button,
+  FormControl,
+  FormHelperText,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { Controller, useForm } from 'react-hook-form';
+import * as yup from 'yup';
+
+import { usePostExchangeApiKey } from '../../../hooks/recording-oracle';
+import FormExchangeSelect from '../../FormExchangeSelect';
+import { ModalError, ModalLoading, ModalSuccess } from '../../ModalState';
+import BaseModal from "../BaseModal";
+
+type Props = {
+  open: boolean;
+  onClose: () => void;
+  exchangeName?: string;
+};
+
+type APIKeyFormValues = {
+  apiKey: string;
+  secret: string;
+  exchange: string;
+};
+
+const validationSchema = yup.object({
+  apiKey: yup.string().required('Required'),
+  secret: yup.string().required('Required'),
+  exchange: yup.string().required('Required'),
+});
+
+const EditApiKeyModal: FC<Props> = ({ open, onClose, exchangeName }) => {
+  const { mutate: postExchangeApiKey, reset: resetMutation, error, isIdle, isPending, isSuccess, isError } = usePostExchangeApiKey();
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    reset,
+  } = useForm<APIKeyFormValues>({
+    resolver: yupResolver(validationSchema),
+    defaultValues: {
+      exchange: '',
+      apiKey: '',
+      secret: '',
+    },
+  });
+
+  useEffect(() => {
+    if (open && exchangeName) {
+      reset({
+        exchange: exchangeName || '',
+        apiKey: '',
+        secret: '',
+      });
+    }
+  }, [open, exchangeName, reset]);
+
+  const handleClose = () => {
+    if (isPending) return;
+    
+    reset();
+    resetMutation();
+    onClose();
+  };
+
+  const onSubmit = (values: APIKeyFormValues) => {
+    postExchangeApiKey({
+      exchangeName: values.exchange,
```

Comment:
Should trim values here as well

Reply:
does not make sense to me. it's a value from backend and it can't be changed

Reply:
@KirillKirill I meant for api and secret below, just pointed on a wrong line, sorry

Reply:
fixed